pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -431,6 +455,40 @@ func (r *ReconcileKogitoApp) newDCForCR(cr *v1alpha1.KogitoApp, serviceBC obuild
 	return depConfig, nil
 }
 
+// newSAforCR returns the ServiceAccount for the Deployment Config containers template
+func newSAforCR(cr *v1alpha1.KogitoApp) (sa corev1.ServiceAccount) {
+	sa = corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.ServiceAccountName,
+			Namespace: cr.Namespace,
+		},
+	}
+	return sa
+}
+
+func newSARoleBinding(cr *v1alpha1.KogitoApp, sa corev1.ServiceAccount) (roles rbacv1.RoleBindingList) {
+	roles = rbacv1.RoleBindingList{}
+	roles.Items = append(roles.Items, rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: sa.Namespace,
+			Name:      fmt.Sprintf("%s-%s", sa.Name, constants.ServiceAccountRole),
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
```

Comment:
is the intention here to give all kogito deployments `view` rights to the cluster? if so, this seems excessive. what is the purpose of this change?

Reply:
this is the role that we've been testing with the service discovery mechanism to access the Core Services API. Sometimes, this services are in other namespaces, that's why we need a `ClusterRole`. Makes sense?